values/decoding/raw_size_string_iterator.go

```diff
@@ -17,21 +17,19 @@ const (
 // raw size encoded string data.
 // TODO(xichen): Get the buffer from bytes pool.
 type rawSizeStringIterator struct {
-	reader     xio.Reader
-	byteReader io.ByteReader // Same as `reader` but has the proper type to save interface conversions in `Next`
+	reader xio.SimpleReadCloser
```

Comment:
Please keep the `byteReader`, this is to save the interface conversion cost at line 43.

Reply:
Where is the interface conversion? We aren't doing any further casting. Also, we're not doing this in any other location just here.

Both `io.ByteReader` and `xio.SimpleReadCloser` ifaces all implement the `ReadByte() (byte, error)` method.

Reply:
Spoke offline, will-readd. Downcasting happens in the `ReadVarint` fn call.